librosa/core/notation.py

```diff
@@ -551,16 +608,39 @@ def key_to_notes(key: str, *, unicode: bool = True) -> List[str]:
 
     if not match:
         raise ParameterError(f"Improper key format: {key:s}")
+    
 
     pitch_map = {"C": 0, "D": 2, "E": 4, "F": 5, "G": 7, "A": 9, "B": 11}
-    acc_map = {"#": 1, "": 0, "b": -1, "!": -1, "♯": 1, "♭": -1}
 
     tonic = match.group("tonic").upper()
     accidental = match.group("accidental")
-    offset = acc_map[accidental]
+
+    offset = sum([ACC_MAP[acc] for acc in accidental])
 
     scale = match.group("scale")[:3].lower()
 
+    multiple = abs(offset)>=2
+
+    #If multiple accidentals, we use recursion, then cycle through so that the enharmonic equivalent of C is at the beginning again.
+
+    if multiple:
+        sign_map = {+1: "♯", -1: "♭"}
+        additional_acc = sign_map[np.sign(offset)]
+        intermediate_notes = key_to_notes(tonic+additional_acc*(abs(offset)-1)+':'+scale)
+        notes = deque([simplify_note(note+additional_acc) for note in intermediate_notes])
+        #Cycle until the equivalent of 'C' is in the first position. This may be a bit inefficient; if additional_acc == +1, then we need to cycle all the way through.
+        while(note_to_degree(notes[0])!=0):
+            notes.appendleft(notes.pop())
```

Comment:
I feel like this part could be simplified a bit.. some ideas:

Since `notes` is already a `deque` object, you could use `notes.rotate` instead of popping and appending left.  If you could additionally locate the index of the zero in `notes` (which, I believe, will always exist), then you could eliminate the while loop entirely by doing `notes.rotate(index)`.

Now, finding the zero in `notes` is tricky the way the data structure currently works, because the `note_to_degree` function only operates on one note at a time so you have to iteratively call it until you hit the right index.  At the same time, I don't think we actually need the deque object here.  If instead, the `simplify_note` and `note_to_degree` functions were vectorized to support ndarrays of strings (as well as plain strings), then you could so something like:

```python
notes = simplify_note(intermediate_notes + additional_acc)  # May need some tweaking...
degrees = note_to_degree(notes)
notes = np.roll(notes, shift=-np.argwhere(degrees == 0)[0])
```
The first line could require some tweaking because broadcast summing strings like that won't work.  However, since you're creating this simplify note function, you could just add `additional_acc` as a supplemental parameter that gets added in on top of whatever else is in `note`.

For an example of this string vectorization business I'm talking about (to support ndarrays of strings) see
https://github.com/librosa/librosa/blob/09e4a622456bbad314e2e6d5c75879f56d728b9b/librosa/core/notation.py#L879

Basically we add a decorator call and a few extra type hints, but the body of the function doesn't change (much).

Reply:
I like the vectorization idea, but I'm having a bit of trouble making it work locally. 
I tried to add this line of code:
```
@vectorize(otypes = [int])
def note_to_degree(key: str) -> int:
```

When I try to call this method, it converts the strings into byte strings, and then the regex matching stops working:

```
TypeError                                 Traceback (most recent call last)
Cell In[4], line 1
----> 1 simplify_note(np.array(['C##','Cb#']))

File [~/Documents/librosa/librosa/core/notation.py:509](https://file+.vscode-resource.vscode-cdn.net/Users/eugenerabinovich/Documents/~/Documents/librosa/librosa/core/notation.py:509), in simplify_note(key, unicode)
    499 def simplify_note(key: str, unicode: bool = True) -> str:
    500     """Take in a note name and simplify by canceling sharp-flat pairs, and doubling accidentals as appropriate.
    501 
    502     >>> librosa.simplify_note('C♭♯')
   (...)
    507 
    508     """
--> 509     match = NOTE_RE.match(key)
    511     if not match:
    512         raise ParameterError(f"Improper key format: {key:s}")

TypeError: cannot use a string pattern on a bytes-like object
```

Is there a way that I can enforce that the array gets processed as an array of `str` objects instead of `bytes`?

Reply:
Hmm... maybe I had this wrong in my head.  A more relevant example might be `note_to_midi`:
https://github.com/librosa/librosa/blob/09e4a622456bbad314e2e6d5c75879f56d728b9b/librosa/core/convert.py#L734-L751

which doesn't use numpy vectorize, but rather handles the unrolling recursively.

Sorry for the confusion on that, it's been a minute since I implemented this stuff.  We do use vectorization when going the other way though:
https://github.com/librosa/librosa/blob/09e4a622456bbad314e2e6d5c75879f56d728b9b/librosa/core/convert.py#L899-L900

Reply:
Ok, I made a first pass at these changes. It seemed to be working fine locally, but I may have lost some test coverage, so I may have to do another pass at adding tests.